### PR TITLE
Add --echo, --offset and --limit MODSI-32

### PR DESCRIPTION
**Pull request by skoczko**
Add --echo, --offset and --limit. Minor cleanup.

### CODE REVIEW

File: client/src/main/java/org/folio/shared/index/client/Client.java [hunk 1]
```diff
@@ -133,20 +170,27 @@ private void sendMarcXmlChunk(XMLStreamReader stream, Promise<Void> promise)  {
...
-    webClient.putAbs(headers.get(XOkapiHeaders.URL) + "/shared-index/records")
-        .putHeaders(headers)
-        .expect(ResponsePredicate.SC_OK)
-        .expect(ResponsePredicate.JSON)
-        .sendJsonObject(request)
-        .onFailure(promise::fail)
-        .onSuccess(x -> sendMarcXmlChunk(stream, promise));
+    if (echo) {
+      out.println(request);
+      Future.<Void>future(x -> sendMarcXmlChunk(stream, promise));
```
Comment (adamdickmeiss) [inline client/src/main/java/org/folio/shared/index/client/Client.java:184]:
```suggestion
      sendMarcXmlChunk(stream, promise);
```

File: client/src/main/java/org/folio/shared/index/client/Client.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package org.folio.shared.index.client;
 
+import static java.lang.System.out;
```
[the remark this answers is not in the document]
Comment (skoczko) [reply client/src/main/java/org/folio/shared/index/client/Client.java:3]: Generally, yes but for help and logging records it looks pretty bad. But I don't mind..

File: client/src/main/java/org/folio/shared/index/client/Client.java [hunk 3]
```diff
@@ -106,22 +134,31 @@ private void sendIso2709Chunk(MarcStreamReader reader, Promise<Void> promise) {
...
-        .expect(ResponsePredicate.SC_OK)
-        .expect(ResponsePredicate.JSON)
-        .sendJsonObject(request)
-        .onFailure(promise::fail)
-        .onSuccess(x -> sendIso2709Chunk(reader, promise));
+    
+    if (echo) {
+      out.println(request);
+      Future.<Void>future(x -> sendIso2709Chunk(reader, promise));
+      return;
```
[the remark this answers is not in the document]
Comment (adamdickmeiss) [reply client/src/main/java/org/folio/shared/index/client/Client.java:139]: Why?

**Review by adamdickmeiss — CHANGES_REQUESTED** [review]
Extra semicon in one place.. OK with system println , but then let's make SQ shut up by adding 


    @java.lang.SuppressWarnings({"squid:S106"})  // using standard output on purpose

**Comment by skoczko** [comment]
@adamdickmeiss I am little confused about the coverage -- I bumped it up to 100% but after refactoring the ReaderProxy it came down -- I would assume the code get exercised the same way, no?

**Comment by skoczko** [comment]
@adamdickmeiss anyway, should I go ahead and merge?

**Review by adamdickmeiss — APPROVED** [review]

**Pull request merged**